### DISCUSSION TITLE
Troubles implementing LDAP configuration

### DISCUSSION THREAD

**Discussion by tom-vo** [netbox-community/netbox#14009] [Help Wanted!]
Hi,

I am trying to implement LDAP authentication, but I encounter a lot of issues.
These are the steps I am doing (following https://docs.netbox.dev/en/stable/installation/6-ldap/):

```
sudo -s
sudo apt install -y libldap2-dev libsasl2-dev libssl-dev
source /opt/netbox/venv/bin/activate
pip3 install django-auth-ldap
sudo sh -c "echo 'django-auth-ldap' >> /opt/netbox/local_requirements.txt"
```

Then made the changes in the config files and did a "systemctl restart netbox".

After that I got this error:

```
Oct  9 10:41:17 host gunicorn[122018]: PermissionError: [Errno 13] Permission denied: '/opt/netbox/venv/lib/python3.8/site-packages/python_ldap-3.4.3.dist-info'
Oct  9 10:41:17 host systemd[1]: netbox.service: Main process exited, code=exited, status=1/FAILURE
```

I found a post suggesting to run " ./upgrade.sh", which I did, after which I got this error:

```
Oct  9 15:30:16 host systemd[7451]: netbox.service: Failed to execute command: Permission denied
Oct  9 15:30:16 host systemd[7451]: netbox.service: Failed at step EXEC spawning /opt/netbox/venv/bin/gunicorn: Permission denied
Oct  9 15:30:16 host systemd[1]: netbox.service: Main process exited, code=exited, status=203/EXEC
```

Anyone a suggestion how to fix this?

**Comment by mathewbishop** [comment]
Hey Tom,

Is this a Netbox instance that you had previously running and working, and then are trying to add LDAP auth to? Or is this a brand new deployment of Netbox that has never successfully run/deployed yet?

<br/>
<br/>

Starting with the first error, it looks like the `gunicorn` user on your Netbox server doesn't have the right permissions to access the LDAP package you just installed. 

To look at the permissions for the netbox packages in that folder that they reside in, run:

`ls -l /opt/netbox/venv/lib/python3.8/site-packages`

Take a look at the output and compare the permissions, ownership, and group of the other packages/files in that directory with the `python_ldap-3.4.3` package and see if they match. Feel free to post the output here, this should give us a place to start.

**Reply by tom-vo** [reply]
Hi,

I solved it by changing the ownership of the netbox folder:
`chown -R netbox:netbox netbox/`

Did I do something wrong? Or is this step lacking in the documentation?

**Reply by candlerb** [reply]
It's not something missing in the documentation, but is probably an artefact of how your `umask` setting is configured on whatever OS you're using.

On most systems you'd find that all the files are owned by root but world-readable, which is fine - except for the directories which Netbox itself needs to write to, which are documented [here](https://docs.netbox.dev/en/stable/installation/3-netbox/#create-the-netbox-system-user).
